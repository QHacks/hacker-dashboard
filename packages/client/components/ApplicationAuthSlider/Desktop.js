import React, { Component } from "react";
import { red, redDark, offWhite, steel } from "../../assets/colors";

class Desktop extends Component {
  changeSelected(index) {
    this.props.changeSelected(index);
  }

  render() {
    const sliderWidth = Math.trunc(100000 / this.props.items.length) / 1000;

    const sliderCss = `
      position: absolute;
      left: ${this.props.selectedItem * sliderWidth}%;
      top: 0;
      height: 44px;
      background-color: ${red};
      border-radius: 44px;
      width: ${sliderWidth}%;
      z-index: 2;
      transition: left 0.8s ease;
    `;

    const navBarCss = `
      display: grid;
      grid-template-columns: repeat(${this.props.items.length}, 1fr);
      width: 66%;
      max-width: 600px;
      margin: 20px auto 35px auto;
      background-color: ${offWhite};
      height: 44px;
      position: relative;
      z-index: 2;
      box-sizing: content-box;
      border: 1px solid ${steel};
      border-radius: 44px;
      button.item {
        text-align: center;
        line-height: 44px;
        text-transform: uppercase;
        cursor: pointer;
        background: none;
        border: none;
        font-size: 18px;
        z-index: 3;
        transition: color 1s ease;
      }
      button.item.selectedItem {
        color: white;
      }
      button.item:not(.selectedItem) {
        color: #4d4d4d;
      }
      button.item:not(.selectedItem):hover {
        color: ${redDark};
      }
    `;

    const items = this.props.items.map((item, i) => (
      <button
        key={item}
        onClick={() => this.changeSelected(i)}
        className={this.props.selectedItem === i ? "selectedItem item" : "item"}
        type="button"
      >
        {item}
      </button>
    ));

    return (
      <div css={navBarCss}>
        {items}
        <div role="none presentation" css={sliderCss} />
      </div>
    );
  }
}

export default Desktop;
