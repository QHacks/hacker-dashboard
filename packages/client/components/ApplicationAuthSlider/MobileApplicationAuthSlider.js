import React, { Component } from "react";
import { redDark, red, steel } from "../../assets/colors";

class Mobile extends Component {
  changeSelected(index) {
    this.props.changeSelected(index);
  }

  render() {
    const items = this.props.items.map((item, i) => (
      <button
        css={`
          display: block;
          text-align: center;
          line-height: 44px;
          text-transform: uppercase;
          cursor: pointer;
          background: none;
          border: none;
          font-size: 18px;
          z-index: 3;
          transition: 0.5s ease;
          border-radius: 44px;
          width: 100%;
          margin: 12px auto;
          max-width: 300px;
          border: 1px solid ${steel};
          ${this.props.selectedItem === i
            ? `color: white; background-color: ${red};`
            : `color: #4d4d4d; &:hover { color: ${redDark}; }
          `}
        `}
        key={item}
        onClick={() => this.changeSelected(i)}
        type="button"
      >
        {item}
      </button>
    ));

    return (
      <div
        css={`
          margin-top: 35px;
        `}
      >
        {items}
        <hr
          css={`
            color: #c5c9d1;
            margin: 52px auto 10px;
          `}
        />
      </div>
    );
  }
}

export default Mobile;
