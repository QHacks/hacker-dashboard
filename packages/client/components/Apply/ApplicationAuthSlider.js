import React, { PureComponent } from "react";
import { css } from "emotion";

class ApplicationAuthSlider extends PureComponent {
  changeSelected(index) {
    this.props.changeSelected(index);
  }

  render() {
    const sliderWidth = Math.trunc(100000 / this.props.items.length) / 1000;

    const sliderCss = css({
      position: "absolute",
      left: `${this.props.selectedItem * sliderWidth}%`,
      top: 0,
      height: "44px",
      backgroundColor: "#C81C2E",
      borderRadius: "44px",
      width: `${sliderWidth}%`,
      zIndex: 2,
      transition: "left 0.8s ease"
    });

    const navBarCss = css({
      display: "grid",
      gridTemplateColumns: `repeat(${this.props.items.length}, 1fr)`,
      width: "66%",
      "@media screen and (max-width: 760px)": { width: "100%" },
      maxWidth: "600px",
      margin: "20px auto 35px auto",
      backgroundColor: "#F6F6F6",
      height: "44px",
      position: "relative",
      zIndex: 2,
      border: "1px solid #e4e4e4",
      borderRadius: "44px",
      "> button.item": {
        textAlign: "center",
        lineHeight: "44px",
        textTransform: "uppercase",
        cursor: "pointer",
        background: "none",
        border: "none",
        fontSize: "18px",
        zIndex: 3,
        transition: "color 1s ease"
      },
      "> button.item.selectedItem": {
        color: "white"
      },
      "> button.item:not(.selectedItem)": {
        color: "#4d4d4d"
      },
      "> button.item:not(.selectedItem):hover": {
        color: "#C81C2E"
      }
    });

    const items = this.props.items.map((item, i) => (
      <button
        key={item}
        onClick={() => this.changeSelected(i)}
        className={this.props.selectedItem === i ? "selectedItem item" : "item"}
      >
        {item}
      </button>
    ));

    return (
      <div className={navBarCss}>
        {items}
        <div role="none presentation" className={sliderCss} />
      </div>
    );
  }
}

export default ApplicationAuthSlider;
