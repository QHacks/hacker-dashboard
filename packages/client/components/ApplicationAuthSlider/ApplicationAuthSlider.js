import React, { Fragment } from "react";
import MediaQuery from "react-responsive";
import { mobileBreakpoint } from "../../assets/constants";
import Desktop from "./Desktop";
import Mobile from "./Mobile";

const ApplicationAuthSlider = (props) => {
  return (
    <Fragment>
      <MediaQuery query={`(max-width: ${mobileBreakpoint})`}>
        <Mobile {...props} />
      </MediaQuery>
      <MediaQuery query={`(min-width: ${mobileBreakpoint})`}>
        <Desktop {...props} />
      </MediaQuery>
    </Fragment>
  );
};

export default ApplicationAuthSlider;
