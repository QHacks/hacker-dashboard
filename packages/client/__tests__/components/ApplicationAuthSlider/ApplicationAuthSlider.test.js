import React from "react";
import ApplicationAuthSlider from "../../../components/ApplicationAuthSlider/ApplicationAuthSlider";
import { shallow } from "enzyme";

describe("ApplicationAuthSlider", () => {
  it("renders correctly", () => {
    expect(shallow(<ApplicationAuthSlider />)).toMatchSnapshot();
  });
});
