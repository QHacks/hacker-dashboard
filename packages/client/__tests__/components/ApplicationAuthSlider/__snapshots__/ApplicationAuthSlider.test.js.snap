// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationAuthSlider renders correctly 1`] = `
<Fragment>
  <MediaQuery
    maxWidth="774px"
    values={Object {}}
  >
    <Mobile />
  </MediaQuery>
  <MediaQuery
    minWidth="775px"
    values={Object {}}
  >
    <Desktop />
  </MediaQuery>
</Fragment>
`;
